### PR TITLE
[pinniped-proxy] Update from structopt to clapv4. Fixes #5357

### LINKED ISSUE #5357: Replace `ansi_term` dependency in Pinniped-proxy
**Summary**
Replace `ansi_term` dependency in the Pinniped-proxy service.

**Background and rationale**
Dependabot is warning that Rust's dependency `ansi_term` is unmaintained. Project hasn't been updated for at least three years.
It is not a direct dependency in `pinniped-proxy`, but a dependency of `clap` v2, which in turn is a dependency of `structopt`.
[Structopt](https://github.com/TeXitoi/structopt) is not maintained either (see their [pinned issue](https://github.com/TeXitoi/structopt/issues/525)).

Dependabot is warning that it is a critical issue.
image

**Description**
A feasible solution is to migrate `structopt` to [`clap` v3](https://github.com/clap-rs/clap). There is a [migration guide here](https://github.com/clap-rs/clap/blob/master/CHANGELOG.md#migrating).

**Acceptance criteria**
`ansi_term` is not used anymore.

### PR DESCRIPTION
**Pull request by absoludity**
Signed-off-by: Michael Nelson <minelson@vmware.com>

### Description of the change

Replaces `structopt` with `clap`

### Benefits

Removes dependencies on now unmaintained packages

### Applicable issues

- fixes #5357

### CODE REVIEW

File: cmd/pinniped-proxy/src/main.rs
Comment (antgamdia) [inline cmd/pinniped-proxy/src/main.rs:34]: I don't really recall right now if the rest of the log traces in the Kubeapps components are being emitted in local time or in UTC time. I mean, not for this PR, but I wanted to bring this up just in case.

File: cmd/pinniped-proxy/src/cli.rs
Comment (antgamdia) [inline cmd/pinniped-proxy/src/cli.rs:17]: nitpick: is `'p'` preferred over `"p"` in rustfmt ?

**Review by antgamdia — APPROVED** [review]
Great, thanks! Happy to see PRs from you again 😊

**Comment by castelblanque** [comment]
Please don't merge it yet, as we are still in the dependencies freeze period.

File: cmd/pinniped-proxy/src/main.rs
Comment (absoludity) [reply cmd/pinniped-proxy/src/main.rs:34]: Should be the same thing (though is not guaranteed to be. Most images are built with UTC as the local TZ, but yep, it may be worth verifying.

File: cmd/pinniped-proxy/src/cli.rs
Comment (absoludity) [reply cmd/pinniped-proxy/src/cli.rs:17]: Actually a different type - `p` is a character type, whereas "p" is effectively a vector of characters, that happens to have a length of 1. I was forced to change this one here because they type had changed in the library so that the short version of the arg is only a single char, I assume :)

**Comment by absoludity** [comment]
> Great, thanks! Happy to see PRs from you again blush

It's nice to be back :)

**Comment by absoludity** [comment]
> Please don't merge it yet, as we are still in the dependencies freeze period.

Thanks for the heads-up, I'm going to mark this as draft just so I don't do so in the morning accidentally :P

**Comment by castelblanque** [comment]
@absoludity The PR can be merged if you want. v2.6.0 has been released and dependencies can be changed now.

**Pull request merged**
